gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -88,6 +92,13 @@
   private final QueryServiceClient queryServiceClient;
   private final int queryServiceRequestTimeout;
   private final AttributeMetadataProvider metadataProvider;
+  private final ExecutorService executorService =
+      Executors.newFixedThreadPool(
+          3, // considering 1 incoming + 2 max outgoing request
```

Comment:
We can make it to `6` for two concurrent requests.

Reply:
Added a common configurable threadpool in https://github.com/hypertrace/gateway-service/pull/121 that can be reused here.

Reply:
will use it here.